extensions/amp-animation/0.1/web-animations.js

```diff
@@ -0,0 +1,363 @@
+            this.context_.resolveTarget(targetSpec) :
+            targetSpec,
+        `Target not found: "${targetSpec}"`);
+    if (this.targets_.indexOf(target) == -1) {
+      this.targets_.push(target);
+    }
+    return target;
+  }
+
+  /**
+   * @param {!Element} target
+   * @param {string} prop
+   * @private
+   */
+  measure_(target, prop) {
+    const index = this.targets_.indexOf(target);
+    if (!this.computedStyleCache_[index]) {
+      this.computedStyleCache_[index] = /** @type {!CSSStyleDeclaration} */ (
+          this.win./*OK*/getComputedStyle(target));
+    }
+    const vendorName = getVendorJsPropertyName(
+        this.computedStyleCache_[index], prop);
+    return this.computedStyleCache_[index].getPropertyValue(vendorName);
+  }
+
+  /**
+   * Merges timing by defaulting values from the previous timing.
+   * @param {!WebAnimationTimingDef} newTiming
+   * @param {!WebAnimationTimingDef} prevTiming
+   * @return {!WebAnimationTimingDef}
+   * @private
+   */
+  mergeTiming_(newTiming, prevTiming) {
+    const duration = newTiming.duration != null ?
+        Number(newTiming.duration) : prevTiming.duration;
+    const delay = newTiming.delay != null ?
+        Number(newTiming.delay) : prevTiming.delay;
+    const endDelay = newTiming.endDelay != null ?
+        Number(newTiming.endDelay) : prevTiming.endDelay;
+    const iterations = newTiming.iterations != null ?
+        Number(newTiming.iterations) : prevTiming.iterations;
+    const iterationStart = newTiming.iterationStart != null ?
+        Number(newTiming.iterationStart) : prevTiming.iterationStart;
+    const easing = newTiming.easing != null ?
+        String(newTiming.easing) : prevTiming.easing;
+    const direction = newTiming.direction != null ?
+        newTiming.direction : prevTiming.direction;
+    const fill = newTiming.fill != null ?
+        newTiming.fill : prevTiming.fill;
+
+    // Validate.
+    if (this.validate_) {
+      user().assert(!isNaN(duration),
+          '"duration" is invalid: %s', newTiming.duration);
+      user().assert(!isNaN(delay),
+          '"delay" is invalid: %s', newTiming.delay);
+      user().assert(!isNaN(endDelay),
+          '"endDelay" is invalid: %s', newTiming.endDelay);
+      user().assert(!isNaN(iterations),
+          '"iterations" is invalid: %s', newTiming.iterations);
+      user().assert(!isNaN(iterationStart),
+          '"iterationStart" is invalid: %s', newTiming.iterationStart);
+      user().assertEnumValue(WebAnimationTimingDirection,
+          /** @type {string} */ (direction), 'direction');
+      user().assertEnumValue(WebAnimationTimingFill,
+          /** @type {string} */ (fill), 'fill');
+    }
+    return {
+      duration,
+      delay,
+      endDelay,
+      iterations,
+      iterationStart,
+      easing,
+      direction,
+      fill,
+    };
+  }
+}
+
+
+/**
+ * Clones an array or an object.
+ * @param {T} x
+ * @return T
+ * @template T
+ */
+function clone(x) {
```

Comment:
I had it there :) Everyone hated it and it was eventually removed. In this case, however, we definitely need it... /cc @jridgewell